### PR TITLE
Twitter clone

### PR DESCRIPTION
**Pull request by DejazmachMolla**
Created a clone of Twitter to practice Authentication and Authorization using devise. The project

- enables registration, log in and logout of users

- allows edit and delete of tweeets only for authorized users

### CODE REVIEW

**Review by LamiaSristy — CHANGES_REQUESTED** [review]
# Status: Change Required. ♻️

Hello @ 🙌

Great work so far. You have used `devise`gem for authentication correctly. 👏 👏 👏

You need to implement the works below to get the project approved!🛠️

Please check my points here: 👇

`[REQUIRED]`

- [x] As a guest you can see only the post's text, not the username of the author.  ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/guest_can_see_only_posts_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication))
ss1

- [x] In `routes.rb` file, It creates the application routes using `resources :posts, only: [:new, :create, :index]` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/posts_routes_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication))
In your case it should be `resources :tweeets, only: [:new, :create, :index]`

- [x] In `tweeets_controller.rb`, the `new` method initializes an empty post `@post = Post.new` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/posts_new_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication))In your case it should be ` @tweeet = Tweeet.new` 

- [x] In `tweeets_controller.rb`, the `index` method doesn't needs `@tweeet = Tweeet.new`.


Please make the changes and ask for a re-review.

Feel free to reach me at @lamiasristy 🙋

File: config/routes.rb [hunk 1]
```diff
@@ -1,4 +1,6 @@
 Rails.application.routes.draw do
+  devise_for :users, :controllers => { registrations: 'registrations' }
   resources :tweeets
```
Comment (LamiaSristy) [inline config/routes.rb]: It creates the application routes using :  `resources :tweeets, only: [:new, :create, :index]` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/posts_routes_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication))

File: app/controllers/tweeets_controller.rb [hunk 2]
```diff
@@ -1,31 +1,31 @@
 class TweeetsController < ApplicationController
-  before_action :set_tweeet, only: %i[ show edit update destroy ]
+  before_action :set_tweeet, only: %i[show edit update destroy]
+  before_action :authenticate_user!, except: %i[index show]
 
   # GET /tweeets or /tweeets.json
   def index
-    @tweeets = Tweeet.all
+    @tweeets = Tweeet.all.order('created_at DESC')
+    @tweeet = Tweeet.new
```
Comment (LamiaSristy) [inline app/controllers/tweeets_controller.rb:8]: Remove  `@tweeet = Tweeet.new`

File: app/controllers/tweeets_controller.rb [hunk 3]
```diff
@@ -1,31 +1,31 @@
...
   # GET /tweeets/1 or /tweeets/1.json
-  def show
-  end
+  def show; end
 
   # GET /tweeets/new
   def new
-    @tweeet = Tweeet.new
+    @tweeet = current_user.tweeets.build
   end
```
Comment (LamiaSristy) [inline app/controllers/tweeets_controller.rb]: ` @tweeet = Tweeet.new`

File: app/controllers/tweeets_controller.rb (hunk 2 above)
Comment (DejazmachMolla) [reply app/controllers/tweeets_controller.rb:8]: Thanks, but I think this is necessary to enable creation of new tweet on the index page. Had the new tweet page been separate from the index page, it wouldn't have been needed as you said.

**Comment by mimipeshy** [comment]
### STATUS: APPROVED 🥳 🍾 

Hey there @DejazmachMolla and @LamiaSristy  :wave: :wave: good job on working on the requested changes. This project is therefore approved. Go ahead and merge. All the best in the next milestones :man_technologist: 

Regards
Peris Ndanu

**Pull request merged**